### PR TITLE
begin moving docs

### PR DESCRIPTION
**Pull request by s-santillan**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### Sara :reminder_ribbon: 

- [x] Check the rm'ed docs in redirects
- [x] Stretch - go through Pro mentions and reduce mentions of "Pro Engine" except in release notes.

### This PR...

- Moves the OSS docs to under Semgrep Code
- Deprecates the Semgrep Pro Engine terminology where possible (the Pro Engine term is still used in the CLI)

### CODE REVIEW

**Comment by s-santillan** [comment]
@chushili1 it may be easier for you to review using the netlify preview: https://deploy-preview-1555--semgrep-docs-prod.netlify.app/docs

File: docs/support.md [hunk 1]
```diff
@@ -58,7 +56,7 @@ through [email](mailto:support@semgrep.com).
 
 Users of Semgrep OSS can log bugs and feature requests in the
 [semgrep](https://github.com/semgrep/semgrep/issues) repository. They can also
-ask questions regarding usage, rollouts, and deployments in the [community Slack
+ask questions regarding usage, roll outs, and deployments in the [community Slack
```
Comment (khorne3) [inline docs/support.md:59]: I think it is "rollsouts"?

**Review by khorne3 — APPROVED** [review]

File: docs/support.md (hunk 1 above)
Comment (s-santillan) [reply docs/support.md:59]: Vale did me dirty. Will add rollouts to the allowlist.

**Comment by s-santillan** [comment]
Verbal approval from requester: 
image

**Pull request merged**